### PR TITLE
Adds uncached functions to VIP RestrictedFunctionsSniff.php.

### LINKED ISSUE #321: Flag wp_get_object_terms() as restricted by VIP
https://vip.wordpress.com/documentation/caching/uncached-functions/

* wp_get_post_terms(), wp_get_post_categories(), wp_get_post_tags(), and wp_get_object_terms()
  * Use get_the_terms() along with wp_list_pluck to extract the IDs

### PR DESCRIPTION
**Pull request by allan23**
Adds uncached functions to VIP.RestrictedFunctionsSniff. This should resolve #321 and includes other uncached functions mentioned in https://vip.wordpress.com/documentation/uncached-functions/

### CODE REVIEW

File: WordPress/Sniffs/VIP/RestrictedFunctionsSniff.php [hunk 1]
```diff
@@ -104,6 +104,64 @@ public function getGroups() {
...
+				'type' => 'error',
+				'message' => '%s is prohibited, please use WP_Query instead.',
+				'functions' => array(
+					'get_children',
+				),
+			),
+
+			'wp_get_object_terms' => array(
+				'type' => 'error',
+				'message' => '%s is prohibited, please use get_the_terms() along with wp_list_pluck to extract the IDs.',
```
Comment (GaryJones) [inline WordPress/Sniffs/VIP/RestrictedFunctionsSniff.php]: Is there a `()` missing here?

File: WordPress/Sniffs/VIP/RestrictedFunctionsSniff.php [hunk 2]
```diff
@@ -104,6 +104,64 @@ public function getGroups() {
...
+				'functions' => array(
+					'wp_get_object_terms'
+				),
+			),
+			
+			'wp_get_post_(categories|tags|terms)' => array(
+				'type' => 'error',
+				'message' => '%s is prohibited, please use the equivalent get_the_* version instead (e.g. get_the_category())',
```
Comment (GaryJones) [inline WordPress/Sniffs/VIP/RestrictedFunctionsSniff.php]: `()` missing here too.

File: WordPress/Sniffs/VIP/RestrictedFunctionsSniff.php [hunk 3]
```diff
@@ -104,6 +104,64 @@ public function getGroups() {
...
+				'type' => 'error',
+				'message' => '%s is prohibited, please use wpcom_vip_get_category_by_slug() instead.',
+				'functions' => array(
+					'get_category_by_slug',
+				), 
+			),
+			
+			'get_cat_ID' => array(
+				'type' => 'error',
+				'message' => '%s is prohibited, please use wpcom_vip_get_term_by() instead',
```
Comment (GaryJones) [inline WordPress/Sniffs/VIP/RestrictedFunctionsSniff.php]: The next few messages (and one a few lines above) are missing full stops.

File: WordPress/Sniffs/VIP/RestrictedFunctionsSniff.php [hunk 4]
```diff
@@ -104,6 +104,64 @@ public function getGroups() {
...
+				),
+			),
+			
+			'wp_get_post_(categories|tags|terms)' => array(
+				'type' => 'error',
+				'message' => '%s is prohibited, please use the equivalent get_the_* version instead (e.g. get_the_category())',
+				'functions' => array(
+					'wp_get_post_terms',
+					'wp_get_post_categories',
+					'wp_get_post_tags'
```
Comment (GaryJones) [inline WordPress/Sniffs/VIP/RestrictedFunctionsSniff.php]: Should add a trailing comma, and can also put into alphabetical order so they are the same order as the parent key.

**Comment by allan23** [comment]
I copied most of the messages from the VIP docs verbatim. I'll see when I can get to working on this.

**Comment by JDGrimes** [comment]
I think that #491 covered part (but maybe not all) of this.

**Comment by westonruter** [comment]
Ah, sorry @allan23. I totally missed your PR. If you see anything missing in my PR, please revise. I also asked @sboisvert to identify anything missing. I uses the latest Uncached Functions listed on VIP docs as of yesterday.

**Pull request closed without merging**